lib/validator/v3/validator.go

```diff
@@ -1333,6 +1387,33 @@ func validateGlobalNetworkPolicy(structLevel validator.StructLevel) {
 		}
 	}
 
+	// Services are only allowed on egress rules.
```

Comment:
Do we not also need to check that services are not set as the source on the egress rules for the GNP?